### PR TITLE
Added program to swap two numbers using bitwise

### PR DESCRIPTION
**Pull request by PuneetTri**
Detailed program to swap two numbers using bitwise operators. Also provided detailed proof of how the swapping happens. This program would be great for beginners to understand how we can use the bitwise operator to swap two numbers, as well as understand how the XOR operator works in general.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Review by mishraabhinn — COMMENTED** [review]
Hey there! Thank you for submitting a PR! 😄 Sadly, this does not meet our requirements. Check out our [repository standards](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md). Please read them carefully and follow them. Let us know if you need any help here or on our [Discord server](https://the-algorithms.com/discord). 🙂

File: bit_manipulation/swapping_two_numbers.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ * More about it: https://betterexplained.com/articles/swap-two-variables-using-xor
+ * @details
+ * This program is part of bitwise manipulation programs, this one
+ * specifically helps us swap two numbers without using an auxillary
+ * variable.
+ * @author [Puneet Tripathi](https://github.com/PuneetTri)
+ */
+
+#include<iostream>  // For standard I/O, in our case printing values of variable 'a' and 'b'
+#include<cassert> // For testcases check
```
Comment (mishraabhinn) [inline bit_manipulation/swapping_two_numbers.cpp:13]:
```suggestion
#include<cassert> // For assert
```

**Review by mishraabhinn — CHANGES_REQUESTED** [review]

**Review by mishraabhinn — COMMENTED** [review]
Thanks for the continuous changes with comments and for bearing with us. Good work. Approved.

**Review by mishraabhinn — APPROVED** [review]

File: bit_manipulation/swapping_two_numbers.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
+/**
+ * @file
+ * @brief Swapping two numbers using Bitwise operator
+ * More about it: https://betterexplained.com/articles/swap-two-variables-using-xor
```
Comment (Panquesito7) [inline bit_manipulation/swapping_two_numbers.cpp:4]: Markdown link formatting should be used here.
```suggestion
 * @brief [Swap two numbers using Bitwise Operator](https://betterexplained.com/articles/swap-two-variables-using-xor) implementation
```

File: bit_manipulation/swapping_two_numbers.cpp [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+ * This program is part of bitwise manipulation programs, this one
+ * specifically helps us swap two numbers without using an auxillary
+ * variable.
+ * @author [Puneet Tripathi](https://github.com/PuneetTri)
+ */
+
+#include<iostream>  // For standard I/O, in our case printing values of variable 'a' and 'b'
+#include<cassert> // For assert
+
+bool bitwiseSwap(int a, int b) {
```
Comment (Panquesito7) [inline bit_manipulation/swapping_two_numbers.cpp:15]: Functions should be documented explaining the use of its parameters, a brief description, and what it returns.

File: bit_manipulation/swapping_two_numbers.cpp [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    // Return true if values swapped or else return false
+    if(temp_a == b && temp_b == a) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+static void test() {
```
Comment (Panquesito7) [inline bit_manipulation/swapping_two_numbers.cpp:71]:
```suggestion
/**
 * @brief Self-test implementations
 * @returns void
 */
static void test() {
```

File: bit_manipulation/swapping_two_numbers.cpp [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    assert(bitwiseSwap(0,53) == true); // TC #2
+    std::cout << "Test-case: #3" << std::endl;
+    assert(bitwiseSwap(622,733) == true); // TC #3
+    std::cout << "Test-case: #4" << std::endl;
+    assert(bitwiseSwap(-8246,-9357) == true); // TC #4
+    std::cout << "Test-case: #5" << std::endl;
+    assert(bitwiseSwap(7,7) == true); // TC #5
+}
+
+int main() {
```
Comment (Panquesito7) [inline bit_manipulation/swapping_two_numbers.cpp:86]:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
int main() {
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Good work! 👍
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/PuneetTri/C-Plus-Plus/actions

File: bit_manipulation/swapping_two_numbers.cpp [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+ * @brief [Swap two numbers using Bitwise Operator](https://betterexplained.com/articles/swap-two-variables-using-xor) implementation
+ * @details
+ * This program is part of bitwise manipulation programs, this one
+ * specifically helps us swap two numbers without using an auxillary
+ * variable.
+ * @author [Puneet Tripathi](https://github.com/PuneetTri)
+ */
+
+#include<iostream>  // For standard I/O, in our case printing values of variable 'a' and 'b'
+#include<cassert> // For assert
```
Comment (Panquesito7) [inline bit_manipulation/swapping_two_numbers.cpp:12]:
```suggestion
#include <iostream>  /// for IO operations
#include <cassert>  /// for assert
```

File: bit_manipulation/swapping_two_numbers.cpp [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+bool bitwiseSwap(int a, int b) {
```
Comment (Panquesito7) [inline bit_manipulation/swapping_two_numbers.cpp:14]:
```suggestion
/**
 * @brief Bit Manipulation algorithms
 * @namespace
 */
namespace bit_manipulation {
bool bitwiseSwap(int a, int b) {
```

File: bit_manipulation/swapping_two_numbers.cpp [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    std::cout << "A: " << a << ", B: " << b << std::endl;
+    std::cout << std::endl; // For better spacing between test-cases
+
+    // Return true if values swapped or else return false
+    if(temp_a == b && temp_b == a) {
+        return true;
+    } else {
+        return false;
+    }
+}
```
Comment (Panquesito7) [inline bit_manipulation/swapping_two_numbers.cpp:68]:
```suggestion
}
}  // namespace bit_manipulation
```

File: bit_manipulation/swapping_two_numbers.cpp [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    std::cout << "Test-case: #1" << std::endl;
+    assert(bitwiseSwap(2,3) == true); // TC #1
+    std::cout << "Test-case: #2" << std::endl;
+    assert(bitwiseSwap(0,53) == true); // TC #2
+    std::cout << "Test-case: #3" << std::endl;
+    assert(bitwiseSwap(622,733) == true); // TC #3
+    std::cout << "Test-case: #4" << std::endl;
+    assert(bitwiseSwap(-8246,-9357) == true); // TC #4
+    std::cout << "Test-case: #5" << std::endl;
+    assert(bitwiseSwap(7,7) == true); // TC #5
```
Comment (Panquesito7) [inline bit_manipulation/swapping_two_numbers.cpp:86]:
```suggestion
    assert(bit_manipulation::bitwiseSwap(2,3) == true); // TC #1
    std::cout << "Test-case: #2" << std::endl;
    assert(bit_manipulation::bitwiseSwap(0,53) == true); // TC #2
    std::cout << "Test-case: #3" << std::endl;
    assert(bit_manipulation::bitwiseSwap(622,733) == true); // TC #3
    std::cout << "Test-case: #4" << std::endl;
    assert(bit_manipulation::bitwiseSwap(-8246,-9357) == true); // TC #4
    std::cout << "Test-case: #5" << std::endl;
    assert(bit_manipulation::bitwiseSwap(7,7) == true); // TC #5
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: bit_manipulation/swapping_two_numbers.cpp [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    */
+
+    std::cout << "After swapping" << std::endl;
+    std::cout << "A: " << a << ", B: " << b << std::endl;
+    std::cout << std::endl; // For better spacing between test-cases
+
+    // Return true if values swapped or else return false
+    if(temp_a == b && temp_b == a) {
+        return true;
+    } else {
```
Comment (krishnacx) [inline bit_manipulation/swapping_two_numbers.cpp:65]: else condition may not be required.

File: bit_manipulation/swapping_two_numbers.cpp (hunk 9 above)
Comment (krishnacx) [reply bit_manipulation/swapping_two_numbers.cpp:86]: Any negative test cases ?

File: bit_manipulation/swapping_two_numbers.cpp [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+    std::cout << "Test-case: #5" << std::endl;
+    assert(bit_manipulation::bitwiseSwap(7,7) == true); // TC #5
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    test(); // Run multiple testcases to test our algorithm
```
Comment (Panquesito7) [inline bit_manipulation/swapping_two_numbers.cpp:100]:
```suggestion
    test(); // Run multiple testcases to test our algorithm
    return 0;
```

File: bit_manipulation/swapping_two_numbers.cpp [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+
+#include <iostream>  /// for IO operations
+#include <cassert>  /// for assert
+
+/**
+ * @brief Bit Manipulation algorithms
+ * @namespace
+ */
+namespace bit_manipulation {
+bool bitwiseSwap(int a, int b) {
```
Comment (Panquesito7) [inline bit_manipulation/swapping_two_numbers.cpp:19]: Still missing function documentation.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**